### PR TITLE
feat: Notifee service events

### LINKED ISSUE #622: Connect/Disconnect events for Network module
Adding new events to the Network module when a peer connects or disconnects.

We can extend the event types like below:

```
const (
	EventTypeConnect    EventType = 1
	EventTypeDisconnect EventType = 2
	EventTypeGossip     EventType = 3
	EventTypeStream     EventType = 4
)
```
Connect/Disconnect events should contain the `peerID` of the connected/disconnected peer.

### PR DESCRIPTION
**Pull request by ZigBalthazar**
## Description

I added `EventTypeConnect` and `EventTypeDisconnect` to the Network events to handle the `Connect` and `Disconnect` events. 

## Related issue(s)

- Fixes #622

### CODE REVIEW

File: network/notifee.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	n.eventChannel <- &ConnectEvent{PeerID: conn.RemotePeer()}
+}
+
+func (n *NotifeeService) Disconnected(_ lp2pnetwork.Network, conn lp2pnetwork.Conn) {
+	n.eventChannel <- &DisconnectEvent{PeerID: conn.RemotePeer()}
+}
+
+func (n *NotifeeService) Listen(_ lp2pnetwork.Network, _ ma.Multiaddr) {
+	// Handle listen event if needed.
+	// TODO: Log
```
Comment (b00f) [inline network/notifee.go:31]: Add log here please. It is very useful in future.
Pass logger to NotifeeService

File: network/notifee.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+// ListenClose is called when your node stops listening on an address.
+func (n *NotifeeService) ListenClose(_ lp2pnetwork.Network, _ ma.Multiaddr) {
+	// Handle listen close event if needed.
```
Comment (b00f) [inline network/notifee.go:36]: Same here

File: network/interface.go [hunk 3]
```diff
@@ -68,6 +74,23 @@ func (*StreamMessage) Type() EventType {
 	return EventTypeStream
 }
 
+type ConnectEvent struct {
```
Comment (b00f) [inline network/interface.go:77]: Let's add a comment here
```
// ConnectEvent represents a peer connection event.
```

File: network/interface.go [hunk 4]
```diff
@@ -26,12 +26,18 @@ func (t TopicID) String() string {
...
+	EventTypeConnect    EventType = 1
+	EventTypeDisconnect EventType = 2
+	EventTypeGossip     EventType = 3
+	EventTypeStream     EventType = 4
 )
 
 func (t EventType) String() string {
 	switch t {
+	case EventTypeConnect:
+		return "connect-msg"
```
Comment (b00f) [inline network/interface.go:38]:
```suggestion
		return "connect"
```

File: network/interface.go [hunk 5]
```diff
@@ -26,12 +26,18 @@ func (t TopicID) String() string {
...
+	case EventTypeDisconnect:
+		return "disconnect-msg"
```
Comment (b00f) [inline network/interface.go:40]:
```suggestion
		return "disconnect"
```

File: network/interface.go (hunk 4 above)
Comment (b00f) [reply network/interface.go:38]: It is not a message actually.

**Review by b00f — CHANGES_REQUESTED** [review]

**Review by themantre — APPROVED** [review]

**Review by themantre — APPROVED** [review]

**Review by b00f — APPROVED** [review]

**Pull request merged**